addons/mrp/models/stock_move.py

```diff
@@ -142,6 +142,7 @@ def _action_cancel(self):
     def _action_confirm(self, merge=True, merge_into=False):
         moves = self.env['stock.move']
         for move in self:
+            move.reference = move.raw_material_production_id.name or move.production_id.name
```

Comment:
action_confirm in mrp no? At them same place than
https://github.com/odoo/odoo/blob/master/addons/mrp/models/mrp_production.py#L608
Why not in https://github.com/odoo/odoo/blob/master/addons/mrp/models/mrp_production.py#L522